packages/next/server/web/sandbox/context.ts

```diff
@@ -314,3 +326,9 @@ Learn more: https://nextjs.org/docs/api-reference/edge-runtime`)
     warnedAlready.add(name)
   }
 }
+
+function decorateUnhandledError(error: any) {
```

Comment:
Why in this case we can't use the `getServerError` directly?

Reply:
When we catch uncaught exceptions or handled rejection, we need to decorate rather the existing error, rather than creating a new one and throwing it.
That's why I had to extract the decoration bit from react-dev-overlay's `nodeStackFrames`